concurrency-limits-core/src/test/java/com/netflix/concurrency/limits/limiter/AbstractPartitionedLimiterTest.java

```diff
@@ -156,4 +164,54 @@ public void setLimitReservesBusy() {
         Assert.assertEquals(1, limiter.getPartition("batch").getInflight());
         Assert.assertEquals(1, limiter.getInflight());
     }
+
+    @Test
+    public void testBypassPartitionedLimiter() {
+
+        AbstractPartitionedLimiter<String> limiter = (AbstractPartitionedLimiter<String>) TestPartitionedLimiter.newBuilder()
+                .partitionResolver(Function.identity())
+                .partition("batch", 0.1)
+                .partition("live", 0.9)
+                .limit(FixedLimit.of(10))
+                .bypassLimitResolver(new ShouldBypassPredicate())
+                .build();
+
+        for (int i = 0; i < 1; i++) {
+            Assert.assertTrue(limiter.acquire("batch").isPresent());
+            Assert.assertEquals(i+1, limiter.getPartition("batch").getInflight());
+            Assert.assertTrue(limiter.acquire("admin").isPresent());
+        }
+
+        for (int i = 0; i < 9; i++) {
+            Assert.assertTrue(limiter.acquire("live").isPresent());
+            Assert.assertEquals(i+1, limiter.getPartition("live").getInflight());
+            Assert.assertTrue(limiter.acquire("admin").isPresent());
+        }
+    }
+
+    @Test
+    public void testBypassSimpleLimiter() {
+
+        SimpleLimiter<String> limiter = (SimpleLimiter<String>) TestPartitionedLimiter.newBuilder()
+                .limit(FixedLimit.of(10))
+                .bypassLimitResolver(new ShouldBypassPredicate())
+                .build();
+
+        int inflightCount = 0;
+        for (int i = 0; i < 5; i++) {
+            Assert.assertTrue(limiter.acquire("request").isPresent());
+            Assert.assertEquals(i+1, limiter.getInflight());
+            inflightCount++;
+        }
+
+        for (int i = 0; i < 15; i++) {
+            Assert.assertTrue(limiter.acquire("admin").isPresent());
+            Assert.assertEquals(inflightCount, limiter.getInflight());
+        }
+
+        for (int i = 0; i < 5; i++) {
+            Assert.assertTrue(limiter.acquire("request").isPresent());
+            Assert.assertEquals(inflightCount+i+1, limiter.getInflight());
+        }
+    }
```

Comment:
Let's also validate that the limiter allows bypass calls even when regular calls are rejected.

Reply:
Updated test